metabase/metabase
# Migrate smart scalar and night mode to support Mantine theming

Closes https://github.com/metabase/metabase/issues/42159

### Description

This PR migrates the smart scalar component that was written in Emotion to use Mantine components [per the RFC](https://www.notion.so/metabase/94-Migrate-a-subset-of-Emotion-components-to-Mantine-to-support-theming-in-React-embedding-SDK-ebf1c71def0745eaa07678c6da31403f), so we can apply theming values accordingly.

### How to verify

First, verify that the migration works as expected.
- Checkout the master branch in the main repo. Open a question and a dashboard. Open the static embedding preview, and change the theme to night. Capture a screenshot.
- Checkout this branch. Do the same as above. Compare the results - they must be the same.

Next, verify that the Mantine theme applies.
- Checkout the `basic-theming-smart-scalar` branch in the demo app.
- Go to `AppProvider.tsx`. Verify that the `text-medium` is set to `white`. You can change `white` to other color for easy verification.
- The color of the smart scalar's description should now follow this PR.

### Demo

Verify that the night-mode embedding should be exactly the same for this branch as `master`.

![CleanShot 2567-05-09 at 19 35 52@2x](https://github.com/metabase/metabase/assets/4714175/e391d650-b808-48de-a8ce-544da4785fc3)

Verify that the demo app's text color changes.

![CleanShot 2567-05-08 at 19 24 22@2x](https://github.com/metabase/metabase/assets/4714175/94c94a2c-ac92-4dd8-89a1-c8b9270d66b8)
